frozen_repos/brian2.diff

```diff
@@ -1,381 +1,66 @@
-         self.main_queue.append(('run_network', (net, run_lines)))
- 
-         # Manually set the cache for the clocks, simulation scripts might
-diff --git a/brian2/devices/cpp_standalone/templates/makefile b/brian2/devices/cpp_standalone/templates/makefile
-index f3a9ca0f..671fd787 100644
---- a/brian2/devices/cpp_standalone/templates/makefile
-+++ b/brian2/devices/cpp_standalone/templates/makefile
-@@ -4,9 +4,8 @@ SRCS = {{source_files}}
- H_SRCS = {{header_files}}
- OBJS = ${SRCS:.cpp=.o}
- OBJS := ${OBJS:.c=.o}
--CC = @g++
- OPTIMISATIONS = {{ compiler_flags }}
--CFLAGS = -c -Wno-write-strings $(OPTIMISATIONS) -I. {{ openmp_pragma('compilation') }} {{ compiler_debug_flags }}
-+CXXFLAGS = -c -Wno-write-strings $(OPTIMISATIONS) -I. {{ openmp_pragma('compilation') }} {{ compiler_debug_flags }}
- LFLAGS = {{ openmp_pragma('compilation') }} {{ linker_flags }} {{ linker_debug_flags }}
- DEPS = make.deps
- 
-@@ -15,17 +14,17 @@ all: $(PROGRAM)
- .PHONY: all clean
- 
- $(PROGRAM): $(OBJS) $(DEPS) makefile
--	$(CC) $(OBJS) -o $(PROGRAM) $(LFLAGS)
-+	$(CXX) $(OBJS) -o $(PROGRAM) $(LFLAGS)
- 
- clean:
- 	{{ rm_cmd }}
- 
- make.deps: $(SRCS) $(H_SRCS)
--	$(CC) $(CFLAGS) -MM $(SRCS) > make.deps
-+	$(CXX) $(CXXFLAGS) -MM $(SRCS) > make.deps
- 	
- ifneq ($(wildcard $(DEPS)), )
- include $(DEPS)
- endif
- 
- %.o : %.cpp makefile
--	$(CC) $(CFLAGS) $< -o $@
-+	$(CXX) $(CXXFLAGS) $< -o $@
-diff --git a/brian2/devices/cpp_standalone/templates/synapses_push_spikes.cpp b/brian2/devices/cpp_standalone/templates/synapses_push_spikes.cpp
-index 99504814..928ab4bf 100644
---- a/brian2/devices/cpp_standalone/templates/synapses_push_spikes.cpp
-+++ b/brian2/devices/cpp_standalone/templates/synapses_push_spikes.cpp
-@@ -26,14 +26,4 @@
-         {{owner.name}}.advance();
-         {{owner.name}}.push({{_eventspace}}, {{_eventspace}}[_num{{eventspace_variable.name}}-1]);
-     }
--
--    {% if profiled %}
--    // Profiling
--    {% if openmp_pragma('with_openmp') %}
--    const double _run_time = omp_get_wtime() -_start_time;
--    {% else %}
--    const double _run_time = (double)(std::clock() -_start_time)/CLOCKS_PER_SEC;
--    {% endif %}
--    {{codeobj_name}}_profiling_info += _run_time;
--    {% endif %}
- {% endblock %}
-diff --git a/brian2/monitors/spikemonitor.py b/brian2/monitors/spikemonitor.py
-index d495a853..f57c95f3 100644
---- a/brian2/monitors/spikemonitor.py
-+++ b/brian2/monitors/spikemonitor.py
-@@ -104,7 +104,7 @@ def __init__(self, source, event, variables=None, record=True,
-         # Some dummy code so that code generation takes care of the indexing
-         # and subexpressions
-         code = ['_to_record_%s = _source_%s' % (v, v)
--                for v in self.record_variables]
-+                for v in sorted(self.record_variables)]
-         code = '\n'.join(code)
- 
-         self.codeobj_class = codeobj_class
 diff --git a/brian2/tests/features/base.py b/brian2/tests/features/base.py
```

Comment:
I just realized that the all the changes in `brian2.diff` coming below here have changed with this PR? This broke our benchmark scripts. I'm opening an issue for this.

@mstimberg Do you have an idea what happened? Since you seem to have pushed some changes here, but not the ones we had before? Could it be that you had an outdated `brian2.diff` to start with? Maybe it wasn't updated by the merge from master?

Reply:
Um, no sorry, not quite sure what happened here. But I had a hard time looking making proper sense of diffs of diffs, so it's entirely possible that I messed this up...